src/app/shared/components/validate-json-or-yaml/component.spec.ts

```diff
@@ -0,0 +1,230 @@
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+
+  describe('Template', () => {
+    it('should render <mat-hint> when isModelValid is true', () => {
+      component.isDataValid = true;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element).toBeTruthy();
+    });
+
+    it('should render <mat-hint> when data is null', () => {
+      component.data = null;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element).toBeTruthy();
+    });
+
+    it('should render correct text inside <mat-hint> when isOnlyYAML is false', () => {
+      component.isDataValid = true;
+      component.isOnlyYAML = false;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element.textContent.trim()).toEqual('Input should be valid JSON or YAML');
+    });
+
+    it('should render correct text inside <mat-hint> when isOnlyYAML is true', () => {
+      component.isDataValid = true;
+      component.isOnlyYAML = true;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element.textContent.trim()).toEqual('Input should be valid YAML');
+    });
+
+    it('should render <mat-error> when data is truthy and isModelValid is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-hint');
+
+      expect(element).toBeFalsy();
+    });
+
+    it('should render <mat-error> when data is truthy and isModelValid is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element).toBeTruthy();
+    });
+
+    it('should render correct text inside <mat-error> when isOnlyYAML is false', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      component.isOnlyYAML = false;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element.textContent.trim()).toEqual('Please enter valid JSON or YAML');
+    });
+
+    it('should render correct text inside <mat-error> when isOnlyYAML is true', () => {
+      component.isDataValid = false;
+      component.data = 'sample text';
+      component.isOnlyYAML = true;
+      fixture.detectChanges();
+
+      const element = fixture.debugElement.nativeElement.querySelector('mat-error');
+
+      expect(element.textContent.trim()).toEqual('Please enter valid YAML');
+    });
+  });
+
+  describe('Component', () => {
```

Comment:
I am not getting the description `Component` what? Which component see other places 

It will be great to full component name to know in while running test case which spec is running.

e.g 
- AddMachineNetworkComponent
- ConstraintsComponent
- RBACComponent

Reply:
Here is the structure of all tests 
<img width="655" alt="image" src="https://user-images.githubusercontent.com/56511988/175511792-25036789-19c6-44ea-ade8-7201b089f790.png">

as you can see, `Template` refers to the HTML angular template and `Component` refers to the Typescript file.

Reply:
structure of unit tests changed